dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1200,11 +1200,102 @@ public boolean isEnabled() {
 
     public final static class TeePrintStream extends PrintStream {
         protected final TrOutputStream trStream;
+        protected final boolean autoFlush;
 
         public TeePrintStream(TrOutputStream trStream, boolean autoFlush) {
             super(trStream, autoFlush);
+            this.autoFlush = autoFlush;
             this.trStream = trStream;
         }
+
+        public void realFlush(String s) {
+            if (autoFlush) {
+                try {
+                    trStream.realFlush();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            if (autoFlush && (s.indexOf('\n') >= 0)) {
+                try {
+                    trStream.realFlush();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        @Override
+        public void println() {
+            super.println();
+            try {
+                trStream.realFlush();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        @Override
+        public void println(String s) {
+            super.print(s);
+            realFlush(s);
+        }
+
+        @Override
+        public void print(String s) {
+            super.print(s);
+            realFlush(s);
+        }
+
+        @Override
+        public void print(boolean b) {
+            super.print(b);
+            String str = b ? "true" : "false";
+            realFlush(str);
+        }
+
+        @Override
+        public void print(char c) {
+            super.print(c);
+            realFlush(String.valueOf(c));
+        }
+
+        @Override
+        public void print(int i) {
+            super.print(i);
+            realFlush(String.valueOf(i));
```

Comment:
another reason to check for \n in the print methods... here you're having to convert i to string even though you know it will never have a \n in it.